src/lib/scheduler/worker.cpp

```diff
@@ -72,7 +72,9 @@ void Worker::operator()() {
     if (!task) {
       // Simple work stealing without explicitly transferring data between nodes.
       auto work_stealing_successful = false;
-      for (auto& queue : scheduler->queues()) {
+      const auto queues = {scheduler->queues().at(static_cast<size_t>(SchedulePriority::High)),
+                           scheduler->queues().at(static_cast<size_t>(SchedulePriority::Normal))};
```

Comment:
This is the part that is causing the problems for Leander. I have no idea what is supposed to be happening here, because it completely contradicts the way the queues are set up:

```
void NodeQueueScheduler::begin() {
  // [...]
  _queues.reserve(_topology->nodes().size());

  for (NodeID q{0}; q < _topology->nodes().size(); q++) {
    auto queue = std::make_shared<TaskQueue>(q);
    _queues.emplace_back(queue);
}
```

If we enable HYRISE_NUMA_SUPPORT but set `max_num_threads` in topology.cpp:16 to 1, the tests should still work.

I would try to fix this myself, but I am not sure of the expected behavior.

Reply:
Good catch, that code was intended to prevent work stealing from `Unstealable` queue, but ended up in the wrong spot. I attempted a fix in f94b398